tests/unit/backend/layers/api/test_curation_api.py

```diff
@@ -1496,6 +1496,72 @@ def test_get_datasets_200(self):
             self.assertEqual(dataset_ids, received_dataset_ids)
 
 
+class TestGetDatasetVersion(BaseAPIPortalTest):
+    def test_get_dataset_version_ok(self):
+        collection = self.generate_published_collection()
+        collection_id = collection.collection_id
+        initial_published_dataset = collection.datasets[0]
+        initial_published_dataset_version_id = collection.datasets[0].version_id
+        published_revision = self.generate_revision(collection_id)
+        published_dataset_revision = self.generate_dataset(
+            collection_version=published_revision,
+            replace_dataset_version_id=initial_published_dataset_version_id,
+            publish=True,
+        )
+        unpublished_revision = self.generate_revision(collection_id)
+        unpublished_dataset_revision = self.generate_dataset(
+            collection_version=unpublished_revision,
+            replace_dataset_version_id=DatasetVersionId(published_dataset_revision.dataset_version_id),
+        )
+
+        headers = self.make_owner_header()
+
+        # get previously published dataset version
+        test_url = f"/curation/v1/dataset_versions/{initial_published_dataset_version_id}"
+        response = self.app.get(test_url, headers=headers)
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(initial_published_dataset_version_id.id, response.json["dataset_version_id"])
+        self.assertEqual(initial_published_dataset.dataset_id.id, response.json["dataset_id"])
+        self.assertEqual(collection_id.id, response.json["collection_id"])
+        self.assertTrue(response.json["explorer_url"].endswith(f"/e/{initial_published_dataset_version_id}.cxg/"))
+
+        # get currently published dataset version
+        test_url = f"/curation/v1/dataset_versions/{published_dataset_revision.dataset_version_id}"
+        response = self.app.get(test_url, headers=headers)
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(published_dataset_revision.dataset_version_id, response.json["dataset_version_id"])
+        self.assertEqual(published_dataset_revision.dataset_id, response.json["dataset_id"])
+        self.assertEqual(collection_id.id, response.json["collection_id"])
+        self.assertTrue(
+            response.json["explorer_url"].endswith(f"/e/{published_dataset_revision.dataset_version_id}.cxg/")
+        )
+
+        # get unpublished dataset version
```

Comment:
I think we want the attempt to fetch an unpublished dataset via `/dataset_version/` to fail.

From the users' perspectives, dataset versions are published entities.